src/realm/obj.cpp

```diff
@@ -1991,53 +1991,10 @@ Obj& Obj::set_collection(ColKey col_key, CollectionType type)
         (col_key.is_list() && type == CollectionType::List)) {
         return *this;
     }
-    checked_update_if_needed();
-    Mixed new_val(0, type);
-
     if (type == CollectionType::Set) {
         throw IllegalOperation("Set nested in Mixed is not supported");
     }
-
-    ArrayMixed arr(_get_alloc());
-    ref_type ref = to_ref(Array::get(m_mem.get_addr(), col_key.get_index().val + 1));
-    arr.init_from_ref(ref);
-    auto old_val = arr.get(m_row_ndx);
-
-    if (old_val != new_val) {
-        CascadeState state;
-        if (old_val.is_type(type_TypedLink)) {
-            remove_backlink(col_key, old_val.get<ObjLink>(), state);
-        }
-        else if (old_val.is_type(type_Dictionary)) {
-            Dictionary dict(*this, col_key);
-            dict.remove_backlinks(state);
-        }
-        else if (old_val.is_type(type_List)) {
-            Lst<Mixed> list(*this, col_key);
-            list.remove_backlinks(state);
-        }
-
-        if (SearchIndex* index = m_table->get_search_index(col_key)) {
-            index->set(m_key, new_val);
-        }
-
-        Allocator& alloc = _get_alloc();
-        alloc.bump_content_version();
-
-        Array fallback(alloc);
-        Array& fields = get_tree_top()->get_fields_accessor(fallback, m_mem);
-        ArrayMixed values(alloc);
-        values.set_parent(&fields, col_key.get_index().val + 1);
-        values.init_from_parent();
-
-        values.set(m_row_ndx, new_val);
-        values.set_key(m_row_ndx, CollectionParent::generate_key(0x10));
-
-        sync(fields);
-
-        if (Replication* repl = get_replication())
-            repl->set(m_table.unchecked_ptr(), col_key, m_key, new_val); // Throws
-    }
+    set(col_key, Mixed(0, type));
```

Comment:
this will always generate the sentinel instruction right? we should do the same for embedded lists and dictionaries.

Reply:
💯